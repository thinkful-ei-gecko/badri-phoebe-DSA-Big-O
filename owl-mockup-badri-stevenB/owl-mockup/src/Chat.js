import React from 'react';
import participants from './participant-list';


export default function(props){
        const chatLogWithParties = props.log.map(event => {
            const party = participants.find(participant => participant.id === event.participantId);
            return {
                ...party,
                ...event
            }
        })
    //{new Date(val.timestamp).toString()}
        //should store in order of time...
    const printJsx = [];
    chatLogWithParties.forEach(val => {
      if(val.onStage && val.type==='join') {
          printJsx.push(<li key={val.id}><b>{val.name}</b> joined.</li>)
      }
      if(val.type==='clap') {
        printJsx.push(<li key={val.id}><b>{val.name}</b> clapped.</li>)
      }
      if(val.type==='leave') {
        printJsx.push(<li key={val.id}><b>{val.name}</b> left.</li>)
    }
    if(val.type==='join-stage') {
        printJsx.push(<li key={val.id}><b>{val.name}</b> joined the stage.</li>)
    }
    if(val.type==='leave-stage') {
        printJsx.push(<li key={val.id}><b>{val.name}</b> left the stage.</li>)
    }
    if(val.type==='raise-hand') {
        printJsx.push(<li key={val.id}><b>{val.name}</b> raised their hand.</li>)
    }
    if(val.type==='thumbs-down') {
        printJsx.push(<li key={val.id}><b>{val.name}</b> gave a thumbs down.</li>)
    }
    if(val.type==='thumbs-up') {
        printJsx.push(<li key={val.id}><b>{val.name}</b> gave a thumbs up. </li>)
    }
    if(val.type==='message') {
        printJsx.push(<li key={val.id}><b>{val.name}</b> {new Date(val.timestamp).toString()} {val.message}</li>)
    }
  
    })
    
    return <div className='chat-box'>Chat: <ul >{printJsx}</ul></div>;
}
