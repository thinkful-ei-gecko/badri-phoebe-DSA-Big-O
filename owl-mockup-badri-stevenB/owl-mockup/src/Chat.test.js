import React from 'react';
import ReactDOM from 'react-dom';
import Chat from './Chat';
import renderer from 'react-test-renderer';


describe ("ChatLog componenet", () => {
    it ("renders without crashing", () => {
        const div = document.createElement('div');
        ReactDOM.render(<Chat log={[1,2,3,4]}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    })
    it ("renders the UI as expected", () => {
        const tree = renderer
        .create(<Chat log={[1,2,3,4]} />)
        .toJSON();
        expect(tree).toMatchSnapshot();
    })
})