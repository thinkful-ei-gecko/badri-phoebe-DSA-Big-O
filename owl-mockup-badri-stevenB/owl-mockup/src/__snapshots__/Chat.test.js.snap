// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatLog componenet renders the UI as expected 1`] = `
<div
  className="chat-box"
>
  Chat: 
  <ul />
</div>
`;
